compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -585,6 +592,37 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         (err, candidates)
     }
 
+    fn get_single_associated_item(
+        &mut self,
+        path: &[Segment],
+        span: Span,
+        source: &PathSource<'_>,
+        filter_fn: &impl Fn(Res) -> bool,
+    ) -> Option<Ident> {
+        if let crate::PathSource::TraitItem(_) = source {
+            let mod_path = &path[..path.len() - 1];
+            if let PathResult::Module(ModuleOrUniformRoot::Module(module)) =
+                self.resolve_path(mod_path, None, false, span, CrateLint::No)
+            {
+                let resolutions = self.r.resolutions(module).borrow();
+                let targets: Vec<_> = resolutions
+                    .iter()
+                    .filter_map(|(key, resolution)| {
+                        resolution
+                            .borrow()
+                            .binding
+                            .map(|binding| binding.res())
```

Comment:
I'm guessing the `add_typo_suggestion` function does something similar to this piece of code. Do you think these two functions can reasonably share code? Or will that just complicate things?

Reply:
😆 funny, but not wrong per se, it is the similarest name after all. Let's do that then!